aspnetcore/performance/performance-best-practices/samples/3.0/Startup22.cs

```diff
@@ -0,0 +1,54 @@
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.DependencyInjection;
+using System.Threading.Tasks;
+
+namespace PBP
+{
+    public class Startup22
+    {
+        public void ConfigureServices(IServiceCollection services)
+        {
+        }
+
+        public void Configure(IApplicationBuilder app)
+        {
+            #region snippet1
+            app.Use(async (context, next) =>
+            {
+                await context.Response.WriteAsync("Hello ");
+
+                await next();
+
+                context.Response.Headers["test"] = "test value";
+            });
+            #endregion
+
+            #region snippet2
+            app.Use(async (context, next) =>
+            {
+                await context.Response.WriteAsync("Hello ");
```

Comment:
Remove this line. With this here, it makes no sense why we're checking `context.Response.HasStarted` since it would almost\* always be true. If only `next()` is called, it remains ambiguous whether or not the response started.

\* `context.Response.HasStarted` could still be false if the response body Stream was replaced before attempting to write to it, but since it's not mentioned anywhere that may have happened, I'd just remove the attempted write.